Remove non RDFJS spec helper functions

martinmaillard opened the pull request:
Inline `isBlankNode`, `isURI`, `isLiteral`, `.lex` and `.uri` to comply with RDFJS spec.

bergos commented on index.js:
```diff
@@ -67,7 +66,7 @@ var SHACLValidator = function() {
 SHACLValidator.prototype.compareNodes = function(node1, node2) {
     // TODO: Does not handle the case where nodes cannot be compared
-    if (node1 && node2 && F.isLiteral(node1) && F.isLiteral(node2)) {
+    if (node1 && node2 && node1.termType === "Literal" && node2.termType === "Literal") {
```
I guess you tried to have a small diff and also from performance perspective the original pattern is maybe a little bit faster for cases where one of the arguments is not given, but it would be more readable to write it like this:

```javascript
node1 && node1.termType === "Literal" && node2 && node2.termType === "Literal"
```

Maybe even rename variables to `a` and `b`. I think that's way more common in compare/sort functions.

I just checked where this method is used. Maybe we can even replace the usage of this method with `a.equals(b)`, but we can also do that in another iteration.

bergos commented on shared/dash.js:
```diff
@@ -304,35 +302,35 @@ function validateOr($value, $or) {
 function validateQualifiedMaxCountProperty($this, $path, $qualifiedValueShape, $qualifiedValueShapesDisjoint, $qualifiedMaxCount, $currentShape) {
 	var c = validateQualifiedHelper($this, $path, $qualifiedValueShape, $qualifiedValueShapesDisjoint, $currentShape);
-	return c <= Number(F.lex($qualifiedMaxCount));
+	return c <= Number($qualifiedMaxCount.value);
```
I guess a `qualifiedMaxCount.termType === 'Literal' && ` should be added. Everywhere else a type check is done before calling `Number` to not call it with `undefined`. That would not be the case, cause `.value` is never `undefined`, but it should not be called with a IRI or Blank Node ID.

martinmaillard replied:
Now the question is, should `validateQualifiedMaxCountProperty` return `true` or `false` if applied to a non literal? And is it even possible?

bergos commented on shared/dash.js:
```diff
@@ -304,35 +302,35 @@ function validateOr($value, $or) {
 function validateQualifiedMinCountProperty($this, $path, $qualifiedValueShape, $qualifiedValueShapesDisjoint, $qualifiedMinCount, $currentShape) {
 	var c = validateQualifiedHelper($this, $path, $qualifiedValueShape, $qualifiedValueShapesDisjoint, $currentShape);
-	return c >= Number(F.lex($qualifiedMinCount));
+	return c >= Number($qualifiedMinCount.value);
```
I guess also here a `qualifiedMinCount.termType === 'Literal' && ` should be added.

bergos commented on shared/rdfquery.js:
```diff
@@ -1019,7 +1017,7 @@ function var2Attr(varName) {
 // where the match object is found.
 function addPathValues(graph, subject, path, set) {
-    if(F.uri(path)) {
+    if(path.value) {
```
This requires to check `path.termType === 'NamedNode'`, cause checking for `.uri` is also a hidden type check.

bergos commented on src/rdfquery.js:
```diff
@@ -1024,7 +1022,7 @@ function var2Attr(varName) {
 // where the match object is found.
 function addPathValues(graph, subject, path, set) {
-    if(F.uri(path)) {
+    if(path.value) {
```
Déjà vu? Is it possible to replace the `shared/*` with `src/*` wherever it is `require`d?

martinmaillard replied:
`src/*` is generated from `shared/*`

bergos commented on shared/dash.js:
```diff
@@ -304,35 +302,35 @@ function validateOr($value, $or) {
 
 function validatePrimaryKeyProperty($this, $path, $uriStart) {
-	if(!F.isURI($this)) {
+	if(!$this.termType === "NamedNode") {
```
Maybe replace it with `!==`?

bergos commented on shared/dash.js:
```diff
@@ -168,7 +166,7 @@ function validateIn($value, $in) {
 
 function validateLanguageIn($value, $languageIn) {
-	if(!F.isLiteral($value)) {
+	if(!$value.termType === "Literal") {
```
Maybe replace it with `!==`?

martinmaillard commented:
Addressing all your comments: I just did an automated search/replace to avoid changing the behavior accidentally. Since "dash.js" and "rdfquery.js" may be deleted in the next iterations, I don't really want to spend time changing their implementation. Hope that makes sense.

bergos commented:
The comments about missing `termType` checks should be added, cause without them information get lost. Rest is OK.
